src/renderer/services/binance/ledger.ts

```diff
@@ -1,27 +1,86 @@
 import * as RD from '@devexperts/remote-data-ts'
+import { Client } from '@xchainjs/xchain-binance'
+import { Address, TxParams } from '@xchainjs/xchain-client'
+import * as E from 'fp-ts/Either'
 import * as FP from 'fp-ts/lib/function'
+import * as O from 'fp-ts/Option'
 import * as Rx from 'rxjs'
-import { catchError, map, startWith } from 'rxjs/operators'
+import { catchError, map, startWith, switchMap } from 'rxjs/operators'
 
-import { Network } from '../../../shared/api/types'
+import { LedgerBNBTxInfo, Network } from '../../../shared/api/types'
 import { observableState } from '../../helpers/stateHelper'
-import { LedgerAddressRD } from '../wallet/types'
+import { XChainClient$ } from '../clients'
+import { ErrorId, LedgerAddressRD, LedgerTxLD, LedgerTxRD } from '../wallet/types'
 import { LedgerService } from './types'
 
-const { get$: ledgerAddress$, set: setLedgerAddressRD } = observableState<LedgerAddressRD>(RD.initial)
+export const createLedgerService = <T extends XChainClient$<Client>>(client$: T): LedgerService => {
+  const { get$: ledgerAddress$, set: setLedgerAddressRD } = observableState<LedgerAddressRD>(RD.initial)
 
-const retrieveLedgerAddress = (network: Network) =>
-  FP.pipe(
-    Rx.from(window.apiHDWallet.getLedgerAddress('BNB', network)),
-    map(RD.fromEither),
-    startWith(RD.pending),
-    catchError((error) => Rx.of(RD.failure(error)))
-  ).subscribe(setLedgerAddressRD)
+  const retrieveLedgerAddress = (network: Network) =>
+    FP.pipe(
+      Rx.from(window.apiHDWallet.getLedgerAddress('BNB', network)),
+      map(RD.fromEither),
+      startWith(RD.pending),
+      catchError((error) => Rx.of(RD.failure(error)))
+    ).subscribe(setLedgerAddressRD)
 
-const createLedgerService = (): LedgerService => ({
-  ledgerAddress$,
-  retrieveLedgerAddress,
-  removeLedgerAddress: () => setLedgerAddressRD(RD.initial)
-})
+  const { get$: ledgerTxRD$, set: setLedgerTxRD } = observableState<LedgerTxRD>(RD.initial)
 
-export { createLedgerService }
+  const handleLedgerPresign = () => {}
+  const handleLedgerVerifySuccess = () => {}
+  const handleLedgerVerifyFailed = () => {
+    setLedgerTxRD(
+      RD.failure({
+        msg: 'Ledger Verification Failed',
+        errorId: ErrorId.SEND_LEDGER_TX
+      })
+    )
+  }
+
+  const ledgerTx$ = (params: TxParams & { sender: Address; network: Network }): LedgerTxLD =>
+    client$.pipe(
+      switchMap((oClient) => (O.isSome(oClient) ? Rx.of(oClient.value) : Rx.EMPTY)),
+      switchMap((client) => Rx.from(client.getBncClient)),
```

Comment:
As mentioned before, we will try to provide needed functions to `xchain-binance`, so that we don't need to access `client$` here.